dot/core/service_test.go

```diff
@@ -257,9 +245,121 @@ func TestProcessBlockResponseMessage(t *testing.T) {
 
 	res, err := s.blockState.GetHeader(header.Hash())
 	require.Nil(t, err)
+	require.Nil(t, err)
+
+	require.Equal(t, header, res)
+
+	for _, blockdata := range bds {
+
+		// test Receipt
+		if blockdata.Receipt.Exists() {
+			receipt, err := s.blockState.GetReceipt(blockdata.Hash)
+			require.Nil(t, err)
+			require.Equal(t, blockdata.Receipt, receipt)
+		}
+
+		// test MessageQueue
+		if blockdata.MessageQueue.Exists() {
+			messageQueue, err := s.blockState.GetMessageQueue(blockdata.Hash)
+			require.Nil(t, err)
+			require.Equal(t, blockdata.MessageQueue, messageQueue)
+		}
+
+		// test Justification
+		if blockdata.Justification.Exists() {
+			justification, err := s.blockState.GetJustification(blockdata.Hash)
+			require.Nil(t, err)
+			require.Equal(t, blockdata.Justification, justification)
+		}
+	}
+}
+
+func TestGetSetReceiptMessageQueueJustification(t *testing.T) {
+	tt := trie.NewEmptyTrie(nil)
+	rt := runtime.NewTestRuntimeWithTrie(t, tests.POLKADOT_RUNTIME, tt)
+
+	kp, err := sr25519.GenerateKeypair()
+	require.Nil(t, err)
+
+	pubkey := kp.Public().Encode()
+	err = tt.Put(tests.AuthorityDataKey, append([]byte{4}, pubkey...))
+	require.Nil(t, err)
+
+	ks := keystore.NewKeystore()
+	ks.Insert(kp)
+
+	cfg := &Config{
+		Runtime:         rt,
+		Keystore:        ks,
+		IsBabeAuthority: false,
+	}
+
+	s := newTestService(t, cfg)
+
+	var genesisHeader = &types.Header{
+		Number:    big.NewInt(0),
+		StateRoot: trie.EmptyHash,
+	}
+
+	hash := common.NewHash([]byte{0})
+	body := optional.CoreBody{0xa, 0xb, 0xc, 0xd}
+
+	parentHash := genesisHeader.Hash()
+	stateRoot, err := common.HexToHash("0x2747ab7c0dc38b7f2afba82bd5e2d6acef8c31e09800f660b75ec84a7005099f")
+	require.Nil(t, err)
+
+	extrinsicsRoot, err := common.HexToHash("0x03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c111314")
+	require.Nil(t, err)
+
+	header := &types.Header{
+		ParentHash:     parentHash,
+		Number:         big.NewInt(1),
+		StateRoot:      stateRoot,
+		ExtrinsicsRoot: extrinsicsRoot,
+		Digest:         [][]byte{},
+	}
+
+	bds := []*types.BlockData{{
+		Hash:          header.Hash(),
+		Header:        header.AsOptional(),
+		Body:          types.NewBody([]byte{}).AsOptional(),
+		Receipt:       optional.NewBytes(false, nil),
+		MessageQueue:  optional.NewBytes(false, nil),
+		Justification: optional.NewBytes(false, nil),
+	}, {
+		Hash:          hash,
+		Header:        optional.NewHeader(false, nil),
+		Body:          optional.NewBody(true, body),
+		Receipt:       optional.NewBytes(true, []byte("asdf")),
+		MessageQueue:  optional.NewBytes(true, []byte("ghjkl")),
+		Justification: optional.NewBytes(true, []byte("qwerty")),
+	}}
+
+	for _, blockdata := range bds {
+
+		err := s.blockState.CompareAndSetBlockData(blockdata)
```

Comment:
why is this test here? it should be in `BlockState` since we are testing `BlockState` functions

Reply:
The test is there because its based on a test that was inside this file. I am not sure I understand your recommendation. Can you be more specific ?

Reply:
I think she means this particular loop only tests block state methods so its not necessary to include in the core service and should be moved to its own test within the state service.

for example, the `// test Receipt` message below checks if the block data has a receipt `Receipt.Exists` using the provided block data, then gets the receipt `blockState.GetReceipt` using block state, then checks the original block data receipt with the receipt retrieved.

This loop should be its own test in block state, which we could provide various block data cases to or leave as is with one block data case for now.

Reply:
done